components/Dropdown.jsx

```diff
@@ -0,0 +1,139 @@
+import React, { PropTypes, Component } from 'react'
+import classNamesBind from 'classnames/bind'
+import styles from '@klarna/ui-css-components/src/components/dropdown.scss'
+import * as programmaticFocus from '../lib/features/programmaticFocus'
+import * as fieldStates from '../lib/features/fieldStates'
+import * as inlinedIcon from '../lib/features/inlinedIcon'
+import { position, size } from '../lib/features/stacking'
+import { handleKeyDown } from '../lib/features/keyboardEvents'
+
+const classNames = classNamesBind.bind(styles)
+
+export default class Dropdown extends Component {
+
+  componentDidMount () {
+    programmaticFocus.maybeFocus(document)(this.props.focus, this.refs.select)
+  }
+
+  componentDidUpdate () {
+    programmaticFocus.maybeFocus(document)(this.props.focus, this.refs.select)
+  }
+
+  render () {
+    const {
+      big,
+      centered,
+      className,
+      disabled,
+      giant,
+      icon,
+      label,
+      loading,
+      onBlur,
+      onChange,
+      onClick,
+      onFocus,
+      options,
+      square,
+      value,
+      ...props
+    } = this.props
+
+    const problem = props.error || props.warning
+    const selectedOption = options && options.find((option) => String(option.value) === String(value))
+
+    const classes = {
+      dropdown: classNames('cui__dropdown--native', {
+          'is-centered': centered,
+          'is-loading': loading,
+          'is-selected': value != null,
+          big,
+          giant,
+          square
+        },
+        fieldStates.getClassName(this.props),
+        position.getClassName(this.props),
+        programmaticFocus.getClassName(this.props),
+        size.getClassName(this.props),
+        className),
+      label: classNames({
+        'cui__dropdown--native__floating-label': problem,
+        'cui__dropdown--native__label': !problem
+      }),
+      currentOption: classNames('cui__dropdown--native__current-option'),
+      select: classNames('cui__dropdown--native__select')
+    }
+
+    return (
+      <div
+        className={classes.dropdown}
+        onClick={onClick}
+      >
+        {
+          <label className={classes.label}>{label}</label>
+        }
+        {
+          selectedOption &&
+            <div className={classes.currentOption}>{selectedOption.label}</div>
+        }
+        <select
+          className={classes.select}
+          disabled={disabled}
+          onBlur={onBlur}
+          onChange={onChange || function () {}}
+          onFocus={onFocus}
+          onKeyDown={handleKeyDown(this.props)}
+          ref='select'
+          {...props}
+        >
+          {
+            options && options.map((attributes) => {
+              return (
+                <option key={attributes.value} {...attributes}>
+                  {attributes.label}
+                </option>
+              )
+            })
+          }
+        </select>
+      </div>
+    )
+  }
+}
+
+Dropdown.optionShape = PropTypes.shape({
```

Comment:
Yup. What do you think about that? I'm not crazy about just spreading instead of explicitly setting them.